### PR TITLE
fix(uninstall): honor global dry-run

### LINKED ISSUE #2952: uninstall ignores global --dry-run and deletes all managed files while reporting a preview
## Summary

`ecc --help` documents a global flag:

```
--dry-run          Preview actions without executing (sets ECC_DRY_RUN=1)
```

The `uninstall` command does not honor it. Running `npx ecc-universal --dry-run uninstall` permanently deleted all 807 ECC-managed files.

## Environment

- ECC 2.2.0 via `npx ecc-universal`
- Linux, Claude Code
- Managed home install: `--profile developer --target claude` (target id `claude-home`)

## Reproduction

1. `npx ecc-universal install --profile developer --target claude`
2. `npx ecc-universal --dry-run uninstall`
3. Output reports:

```
Uninstall summary:

- claude-home
  Status: UNINSTALLED
  Install-state: /home/<user>/.claude/ecc/install-state.json
  Removed paths: 807
```

4. `~/.claude/{skills,commands,agents,rules,hooks}` and `ecc/install-state.json` are gone.

## Expected

No filesystem changes. Output phrased as a preview (e.g. `Would remove: 807 paths`).

## Actual

All managed files deleted. A subsequent real `uninstall` reports:

```
No ECC install-state files found for the current home/project context.
```

## Contributing factor

The output is written in the past tense — `Status: UNINSTALLED`, `Removed paths: 807` — so a dry run is textually indistinguishable from a real one. A user reading the output carefully still gets no signal that deletion occurred.

## Notes

- `ecc help uninstall` lists a subcommand-level `[--dry-run]`. That form was not tested; the global prefix form is what failed.
- Suggested fix: honor `ECC_DRY_RUN` in `uninstall`, or have the CLI reject the global flag for destructive subcommands rather than silently ignoring it.
- Preview output should use conditional wording ("would remove") so a dry run cannot be mistaken for a completed one.

## Severity

Data loss. Recovery required a full reinstall.

### PR DESCRIPTION
**Pull request by wellkilo**
## What Changed

- Honor the global `ECC_DRY_RUN=1` contract in the uninstall command.
- Reject invalid `ECC_DRY_RUN` values before any uninstall discovery or mutation.
- Add an end-to-end regression for `ecc --dry-run uninstall` covering both JSON and human-readable output.
- Verify that managed files and install-state remain untouched, including the legacy Codex cleanup path.
- Isolate CLI tests from inherited dry-run and Codex-home environment values.

## Why This Change

The top-level CLI converts a leading `--dry-run` into `ECC_DRY_RUN=1` before dispatching the subcommand. The uninstall parser only inspected its own argv, so `ecc --dry-run uninstall` silently performed a real uninstall. This fixes the destructive behavior reported in #2952.

Fixes #2952

## Testing Done

- [x] End-to-end dry-run behavior tested with isolated temporary HOME and project directories
- [x] Focused regression suites pass: 128/128
- [x] Node syntax checks, focused ESLint, and `git diff --check` pass
- [x] `scripts/uninstall.js` focused coverage: 86.73% lines, 75.38% branches
- [x] Edge cases cover install-state and legacy Codex cleanup paths, JSON and human output, invalid environment values, and environment isolation
- [ ] Full `npm test` is completely green locally: 4144/4161 passed; the remaining existing failures are 15 validator-fixture failures, one sandboxed HOME write, and one Plan Canvas SSE timeout, all outside this diff. The package-manager test passes 116/116 with an isolated HOME, and the same Plan Canvas timeout reproduces on `origin/main`.

## Type of Change

- [x] `fix:` Bug fix

## Security & Quality Checklist

- [x] No secrets or API keys committed
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## Documentation

- [x] No documentation change required; this restores the already documented global `--dry-run` contract

### CODE REVIEW

File: tests/scripts/ecc.test.js [hunk 1]
```diff
@@ -3,34 +3,40 @@
...
+  }
 
   if (typeof envOverrides.HOME === 'string' && !('USERPROFILE' in envOverrides)) {
-    envOverrides.USERPROFILE = envOverrides.HOME;
+    env.USERPROFILE = envOverrides.HOME;
   }
 
   if (typeof envOverrides.USERPROFILE === 'string' && !('HOME' in envOverrides)) {
-    envOverrides.HOME = envOverrides.USERPROFILE;
+    env.HOME = envOverrides.USERPROFILE;
```
[the remark this answers is not in the document]
Comment (wellkilo) [reply tests/scripts/ecc.test.js]: Addressed in 179d4a4e. Both test helpers now construct isolated subprocess environments with filtered copies and object spreads, without deleting or assigning properties after construction. Focused ESLint and the CLI/uninstall regression suites pass.

File: tests/scripts/ecc.test.js [hunk 2]
```diff
@@ -153,6 +153,79 @@ function main() {
...
+            strategy: 'preserve-relative-path',
+            ownership: 'managed',
+            scaffoldOnly: false,
+            contentSha256: crypto.createHash('sha256').update(managedContent).digest('hex'),
+          }],
+        }));
+
+        const jsonResult = runCli(['--dry-run', 'uninstall', '--target', 'cursor', '--json'], {
+          cwd: projectRoot,
+          env: { HOME: homeDir },
```
[the remark this answers is not in the document]
Comment (wellkilo) [reply tests/scripts/ecc.test.js:202]: This test intentionally keeps the public `ecc --dry-run uninstall` invocation because #2952 is specifically a regression in the top-level flag-to-environment propagation path. Replacing the flag with a pre-set environment variable would bypass `ecc.js` and could let that regression return unnoticed. The child-command environment contract is covered separately by the legacy cleanup test using `ECC_DRY_RUN=1`; the helper also strips inherited values so this top-level test cannot pass accidentally.
